pkg/cmd/extension/command.go

```diff
@@ -293,19 +293,35 @@ func NewCmdExtension(f *cmdutil.Factory) *cobra.Command {
 				Use:   "install <repository>",
 				Short: "Install a gh extension from a repository",
 				Long: heredoc.Docf(`
-					Install a GitHub repository locally as a GitHub CLI extension.
+					Install a GitHub CLI extension from a GitHub or local repository.
 
-					The repository argument can be specified in %[1]sOWNER/REPO%[1]s format as well as a full URL.
+					For GitHub repositories, the repository argument can be specified in %[1]sOWNER/REPO%[1]s format or as a full repository URL.
 					The URL format is useful when the repository is not hosted on github.com.
 
-					To install an extension in development from the current directory, use %[1]s.%[1]s as the
-					value of the repository argument.
+					For local repositories, often used while developing extensions, use %[1]s.%[1]s as the
+					value of the repository argument. Note the following:
+
+					- After installing an extension from a locally cloned repository, the GitHub CLI will
```

Comment:
Do you think it's worth doc-ing how this is different on windows?

Reply:
@jtmcg and I [also wondered about that](https://github.com/cli/cli/pull/9933#discussion_r1847049096) but landed on it being an implementation detail that may confuse the docs if we try to document it. 

What do you think?

Reply:
@williammartin and I talked about this and decided to add a hint to the help text without fully documenting the differences.

Reply:
Resolved in 32abca8c381c34a349d7d829d2fcfda2a4da91e6